### ISSUE TITLE
coverity/62618

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#1748]
Observation from @aktau rescued from https://github.com/neovim/neovim/issues/760#issuecomment-44091948 :

---

```
*** CID 62618:  Wrong sizeof argument  (SIZEOF_MISMATCH)
/src/nvim/eval.c: 18057 in func_dump_profile()
18050       ufunc_T     **sorttab;
>>>     CID 62618:  Wrong sizeof argument  (SIZEOF_MISMATCH)
>>>     Passing argument "216UL /* sizeof (ufunc_T) */ * todo" to function "xmalloc" and then casting the return value to "ufunc_T **" is suspicious.
18057       sorttab = xmalloc(sizeof(ufunc_T) * todo);
```

Ah, the power of coverity ;). After looking at this function it seems pretty clear that this should've been ` sorttab = xmalloc(sizeof(ufunc_T *) * todo);`. Another yay for using `sizeof(var)`: `sorttab = xmalloc(sizeof(*sorttab) * todo);`

Not that it's that important as we don't use profiling and eval will go away, but still, these errors slip by oh-so-easily.

**Comment by oni-link** [comment]
Fix submitted in #2587.

**Comment by Pyrohh** [comment]
Coverity says they're fixed, so I'll close this.
